src/lib/layer-manager.js

```diff
@@ -428,3 +537,33 @@ function layerName(layer) {
   }
   return !layer ? 'null layer' : 'invalid layer';
 }
+
+/**
+ * Get mouse position {x, y} relative to the container
+ */
+function getPos(event) {
+  const {srcEvent, center, rootElement} = event;
+  let x;
+  let y;
+
+  if (center) {
+    // processed pointer input, supports touch
+    x = center.x;
+    y = center.y;
+  } else if (srcEvent) {
+    // fallback
+    x = srcEvent.clientX;
+    y = srcEvent.clientY;
+  }
+  if (!Number.isFinite(x)) {
+    // srcEvent.clientX may be undefined if it's not from a mouse
+    return null;
+  }
+
+  const rect = rootElement.getBoundingClientRect();
```

Comment:
I am not an expert on these things, but folks who seem to be have advised me in the past to avoid calling `getBoundingClientRect` on any regular basis, as it can trigger an expensive [reflow](https://gist.github.com/paulirish/5d52fb081b3570c81e3a) in the browser.

My understanding was that events have `offsetX` and `offsetY` or similar parameters that contain relative coordinates so that this can be handled in a more lightweight fashion.